Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs

```diff
@@ -0,0 +1,228 @@
+                                                   Name = @"\ApplicationInsights\Dependency Call Duration",
+                                                   Value = Round(sample.AIDependencyCallDurationAveInMs),
+                                                   Weight = sample.AIDependencyCalls
+                                               },
+                                           new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Dependency Calls Failed/Sec",
+                                                   Value = Round(sample.AIDependencyCallsFailedPerSecond),
+                                                   Weight = 1
+                                               },
+                                           new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Dependency Calls Succeeded/Sec",
+                                                   Value = Round(sample.AIDependencyCallsSucceededPerSecond),
+                                                   Weight = 1
+                                               },
+                                            new MetricPoint
+                                               {
+                                                   Name = @"\ApplicationInsights\Exceptions/Sec",
+                                                   Value = Round(sample.AIExceptionsPerSecond),
+                                                   Weight = 1
+                                               }
+                                       };
+
+                metricPoints.AddRange(sample.PerfCountersLookup.Select(counter => new MetricPoint { Name = counter.Key, Value = Round(counter.Value), Weight = 1 }));
+
+                var dataPoint = new MonitoringDataPoint
+                                    {
+                                        Version = this.version,
+                                        InstrumentationKey = instrumentationKey,
+                                        Instance = this.instanceName,
+                                        Timestamp = sample.EndTimestamp,
+                                        Metrics = metricPoints.ToArray()
+                                    };
+
+                monitoringPoints.Add(dataPoint);
+            }
+
+            var serializer = new DataContractJsonSerializer(typeof(MonitoringDataPoint[]));
+            serializer.WriteObject(bodyStream, monitoringPoints.ToArray());
+
+            bodyStream.Position = 0;
+
+            return bodyStream;
+        }
+        
+        private HttpWebResponse SendRequest(string httpVerb, string path, MemoryStream body, bool enableRetry)
+        {
+            var requestUri = string.Format(CultureInfo.InvariantCulture, "{0}/{1}", this.ServiceUri.AbsoluteUri.TrimEnd('/'), path.TrimStart('/'));
```

Comment:
There is no good way to concatenate URI's (as outlined in many online discussions, e.g. http://stackoverflow.com/questions/372865/path-combine-for-urls). Both Uri constructor and other ways (including UriBuilder) don't fully solve the problem, they still depend on whether there's a slash at the beginning of the relative path, and the end of the absolute one. Since I need to Trim() anyway, I'd rather form the URI manually.
I have changed one of the Trims to only happen once.